index.js

```diff
@@ -2,46 +2,54 @@ const fs = require("fs");
 const readline = require("readline");
 const Amadeus = require("amadeus");
 const dotenv = require("dotenv");
+const { parse } = require("path");
+
 
 dotenv.config();
 
 const ORIGINS_FILE = process.argv[2]; // Get path to file containing origin airport codes from command-line arguments
-const DESTINATION = process.argv[3]; // Get destination airport code from command-line arguments
+const DESTINATIONS_FILE = process.argv[3]; // Get destination airport code from command-line arguments
 const DEPARTURE_DATE = process.argv[4]; // Get departure date from command-line arguments
 const RETURN_DATE = process.argv[5]; // Get return date from command-line arguments
-
 const amadeus = new Amadeus({
   clientId: process.env.AMADEUS_API_KEY,
   clientSecret: process.env.AMADEUS_API_SECRET,
 });
 
-async function getAirfarePrice(origin, pricesCache) {
-  if (pricesCache[origin] !== undefined) {
+async function getAirfarePrice(origin, destination, pricesCache) {
+  const cacheKey = `${origin}-${destination}`;
+  if (pricesCache[cacheKey] !== undefined) {
     console.log(
-      `Using cached price from ${origin} to ${DESTINATION} on ${DEPARTURE_DATE} - ${RETURN_DATE}: $${pricesCache[origin]}`
+      `Using cached price from ${origin} to ${destination} on ${DEPARTURE_DATE} - ${RETURN_DATE}: $${pricesCache[cacheKey]}`
     );
-    return pricesCache[origin];
+    return pricesCache[cacheKey];
   }
+  console.log(
+    `Checking prices from ${origin} to ${destination}`
+  );
+  if(origin == destination){
+      return 0; //if origin and location are the same we can skip the check and return $0
+  } 
   try {
     const response = await amadeus.shopping.flightOffersSearch.get({
       originLocationCode: origin,
-      destinationLocationCode: DESTINATION,
+      destinationLocationCode: destination,
       departureDate: DEPARTURE_DATE,
       returnDate: RETURN_DATE,
       adults: 1,
       currencyCode: "USD",
-      max: 3,
+      max: 1, // Only fetch the cheapest flight
     });
     if (response?.data && response.data.length > 0) {
       const price = response.data[0].price.total;
-      console.log(
-        `Airfare price from ${origin} to ${DESTINATION} on ${DEPARTURE_DATE} - ${RETURN_DATE}: $${price}`
-      );
-      pricesCache[origin] = price - 0;
-      return pricesCache[origin];
+   //   console.log(
+   //     `Airfare price from ${origin} to ${destination} on ${DEPARTURE_DATE} - ${RETURN_DATE}: $${price}`
+   //   );
```

Comment:
@Crixu curious why you commented this, was it too much noisy?

Reply:
Oh yeah I commented it out as I saved the output as CSV but I guess most people would like this kind of process reporting